### PR TITLE
Add statefulness to the absolute price agent

### LINKED ISSUE #31: Agents persistence
Persist the agents' state across restarts.

Limit persistence time, as the longer AA is down, the further away we get from the current market conditions, at which point it is preferable to restart the training from scratch.

### PR DESCRIPTION
**Pull request by aasseman**
Refer to individual commit messages for details.

### CODE REVIEW

**Review by tumayt — COMMENTED** [review]
I  think agents should have API calls to init schema, restore and save for their state. As autoagora should be agnostic to the internals of agents. Also SqlAlchemy may be useful for clean DDL and queries.
If this is a temporary solution till agents v2, LGTM.

**Comment by aasseman** [comment]
> I think agents should have API calls to init schema, restore and save for their state. As autoagora should be agnostic to the internals of agents.

I agree, but I wanted a solution now and not get bogged down in the complexity of setting up the proper thing :neutral_face:

**Comment by aasseman** [comment]
Had to wait for https://github.com/semiotic-ai/autoagora-agents/releases/tag/v1.2.1 to fix some issues.
Also, rebased onto latest master.

File: autoagora/main.py [hunk 1]
```diff
@@ -35,6 +36,21 @@ async def allocated_subgraph_watcher():
         (args.relative_query_costs_exclude_subgraphs or "").split(",")
     )
 
+    try:
+        pgpool = await asyncpg.create_pool(
+            host=args.postgres_host,
```
Comment (tumaysem) [inline autoagora/main.py:41]: from my previous experience, pool size should be controlled (from args ?)
https://magicstack.github.io/asyncpg/current/api/index.html#asyncpg-api-pool

Comment (aasseman) [reply autoagora/main.py:41]: Yes, you're right, I should make this a user setting.
In general, what do you think of limiting to a single connection by default? None of the code paths are extremely time-sensitive, and I don't expect cyclical dependencies.

Comment (aasseman) [reply autoagora/main.py:41]: https://github.com/semiotic-ai/autoagora/pull/54/commits/096087fbc4ec01a763737d6b4c2714a6da12d816

**Review by tumayt — APPROVED** [review]
LGTM

**Pull request merged**